ui/src/app/settings/views/Scripts/ScriptsUpload.js

```diff
@@ -67,15 +67,32 @@ const ScriptsUpload = ({ type }) => {
       const acceptedFile = acceptedFiles[0];
       const scriptName = pathParse(acceptedFile.path).name;
       const reader = new FileReader();
+
       reader.onabort = () => {
         dispatch(messages.add("Reading file aborted.", "negative"));
       };
       reader.onerror = () => {
         dispatch(messages.add("Error reading file.", "negative"));
       };
+
+      let hasMetadata = false;
       reader.onload = () => {
         const binaryStr = reader.result;
-        setScript({ name: scriptName, script: binaryStr });
+        try {
+          const scriptArray = binaryStr.split("\n");
+          scriptArray.forEach(line => {
+            if (line.includes("--- Start MAAS 1.0 script metadata ---")) {
+              hasMetadata = true;
+            }
+          });
+        } catch {
+          console.error("Unable to parse script for metadata.");
```

Comment:
Shouldn't this be an error notification?

Reply:
Not in this case no, if the metadata can't be parsed we still attempt a normal upload.